### PR TITLE
(PDB-2707) Update our documentation for PuppetDB 4.1 release

### PR DESCRIPTION
**Pull request by ajroetker**

### CODE REVIEW

**Comment by ajroetker** [comment]
@senior @rbrw added the release notes :)

File: documentation/release_notes.markdown [hunk 1]
```diff
@@ -14,6 +14,82 @@ canonical: "/puppetdb/latest/release_notes.html"
 [metrics]: ./api/metrics/v1/changes-from-puppetdb-v3.html
 [pqltutorial]: ./api/query/tutorial-pql.html
 
+4.1.0
+-----
+
+PuppetDB 4.1.0 is a backward-compatible feature release that adds packages for
+Ubuntu Xenial - 16.04, adds preliminary sorry for HUP signal handling, and
```
Comment (senior) [inline documentation/release_notes.markdown]: Probably a freudian slip here - s/sorry/support/

File: documentation/release_notes.markdown [hunk 2]
```diff
@@ -14,6 +14,82 @@ canonical: "/puppetdb/latest/release_notes.html"
...
+  ([PDB-2475](https://tickets.puppetlabs.com/browse/PDB-2475)).
+
+* Allow configuration of AciveMQ Broker's memoryLimit. For PuppetDB instances
+  with larger amount of memory and heavy load, this can improve performance.
+  More information in the [config docs][./configure.html#memory-usage]
+  ([PDB-2726](https://tickets.puppetlabs.com/browse/PDB-2726)).
+
+* Preliminary support for HUP signal handling. Note that due to
+  [AMQ-5263][https://issues.apache.org/jira/browse/AMQ-5263] there is a
+  possibility of a crash when HUPed. We've observed this typically happens when
```
Comment (senior) [inline documentation/release_notes.markdown]: I think I'll remove the "typically" here. It makes it sound like it happens more than it really does.

**Comment by senior** [comment]
This looks good to me. I'll merge this via the command line with a fixup for the two comments I had a the version bump commit

**Comment by senior** [comment]
This was manually merged https://github.com/puppetlabs/puppetdb/commit/67091858aa2370495d2f51421407a2520614f53d

**Pull request closed without merging**